windows/src/support/kmdevlink/UfrmOpenCRMRecord.pas

```diff
@@ -74,28 +89,131 @@ function RepoFullNameToShortName(name: string): string;
   Result := name;
 end;
 
-procedure TfrmOpenCRMRecord.editSearchForChange(Sender: TObject);
+function SearchTextToQuery(s: string): TIssueQuery;
 var
-  s: string;
   parts: TArray<String>;
-  repo: string;
+begin
+  Result.repo := 'keyman';
+  Result.issueNumber := 0;
+  Result.searchString := '';
+
+  s := s.Trim;
+  parts := s.Split(['#']);
+  if Length(parts) = 0 then
+  begin
+    Exit;
+  end
+  else if Length(parts) = 1 then
+  begin
+    Result.issueNumber := StrToIntDef(s, 0);
+    if IntToStr(Result.issueNumber) <> s then
+    begin
+      Result.issueNumber := 0;
+      Result.searchString := s;
+    end;
+  end
+  else
+  begin
+    Result.repo := RepoShortNameToFullName(parts[0]);
+    Result.issueNumber := StrToIntDef(parts[1], 0);
+    if IntToStr(Result.issueNumber) <> parts[1] then
+    begin
+      Result.issueNumber := 0;
+      Result.searchString := parts[1];
+    end;
+  end;
+end;
+
+type
+  TMyClipboard = class(TClipboard);
+
+procedure TfrmOpenCRMRecord.cmdCopyHTMLClick(Sender: TObject);
+var
+  c: TMyClipboard;
+  m: TMemoryStream;
+  ss: TStream;
+  displayRepo, html, s: string;
+  iq: TIssueQuery;
+  CF_HTML: UINT;
+  header: string;
+const
+  // https://learn.microsoft.com/en-us/windows/win32/dataxchg/html-clipboard-format
+  // yeesh what a format
+  header_template =
+    'Version:0.9'#$D#$A+
+    'StartHTML:%0.09d'#$D#$A+
+    'EndHTML:%0.09d'#$D#$A+
+    'StartFragment:%0.09d'#$D#$A+
+    'EndFragment:%0.09d'#$D#$A;
+  start_fragment =  '<!--StartFragment -->';
+  end_fragment = '<!--EndFragment -->';
+  context_start = '<html>'#$D#$A'<body>'#$D#$A;
+  context_end = #$D#$A'</body>'#$D#$A'</html>';
+begin
+  iq := SearchTextToQuery(SearchText);
+  if iq.repo = 'keyman' then
+    displayRepo := ''
+  else
+    displayRepo := iq.repo;
+  html := Format('<a href="https://github.com/keymanapp/%0:s/issues/%1:d">%2:s#%1:d</a>', [
```

Comment:
It's a shame this can't use `SOpenIssueURL`

Reply:
In theory it could but cost/benfit is not worth it